packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,380 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'toggleable.dart';
+
+// Examples can assume:
+// bool _throwShotAway = false;
+// late StateSetter setState;
+
+/// A macOS style checkbox.
+///
+/// The checkbox itself does not maintain any state. Instead, when the state of
+/// the checkbox changes, the widget calls the [onChanged] callback. Most
+/// widgets that use a checkbox will listen for the [onChanged] callback and
+/// rebuild the checkbox with a new [value] to update the visual appearance of
+/// the checkbox.
+///
+/// The checkbox can optionally display three values - true, false, and null -
+/// if [tristate] is true. When [value] is null a dash is displayed. By default
+/// [tristate] is false and the checkbox's [value] must be true or false.
+///
+/// Note: in the Apple ecosystem, checkboxes are encouraged to only be used in
+/// macOS, not iOS. If a multi-selection component is needed, iOS encourages the
+/// developer to use Switches, or [CupertinoSwitch] instead, or find a creative
+/// custom solution.
```

Comment:
I'm not confident on the wording here. I would like to warn developers of Apple's guidance, without discouraging them from doing what they would like to do.